mitodl/open-discussions
# Cc/frontpage round 2

#### Pre-Flight checklist


#### What are the relevant tickets?
#3630 

#### What's this PR do?
This PR makes the infinite-corridor front page match https://projects.invisionapp.com/d/main#/console/22082155/468411241/preview, in particular:
- new header
- nicer search box
- field link list
- styling 

#### How should this be manually tested?
1. *If you want the thumbnail photos to actually work, get `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and `AWS_STORAGE_BUCKET_NAME` from rc.
2. Create some fields via admin panel at http://cac.od.odl.local:8063/admin/channels_fields/fieldchannel/
3. visit http://cac.od.odl.local:8063/infinite ... you should see the homepage.
4. Check that clicking a link goes to the field page (which is a placeholder right now)
5. Check that the searchbox works as expected
6. **Check /learn:** Visit http://cac.od.odl.local:8063/learn and http://cac.od.odl.local:8063/learn/search and check that the searchbox still works correctly. (It's now using the new search input from).

#### Context
The biggest change here is related to the search input. Previously `ol-search-ui` had a `Searchbox` component which `/learn` was using. That component was our custom-built searchbox with icons at start/end. `Searchbox` was kind of annoying because it (A) some container divs and (B) also the search text input. So I broke it into `Searchbox` and `SearchInput`.

Putting icon buttons at the start/end of a text field is nontrivial, so seems best to leave this to a library....So the new SearchInput uses [`@mui`](https://www.npmjs.com/package/@mui/material) ... "Material UI". Open-discussions was already using uses Material UI under its old name, `@material`. (the name changed to `@mui` in  .... v5 iirc). 

#### Screenshots (if appropriate)

<img width="400" alt="infinite_homepage_desktop" src="https://user-images.githubusercontent.com/9010790/179822200-c202a9c4-8b87-41ef-8c8c-6bcebb3a60e9.png"> <img width="200" alt="infinite_homepage_mobile" src="https://user-images.githubusercontent.com/9010790/179822380-074598fa-53a3-4214-b7e9-067cab8c9afb.png">
